core/src/main/java/org/mskcc/cbio/portal/dao/DaoGene.java

```diff
@@ -67,45 +67,58 @@ private static synchronized int getNextFakeEntrezId() throws DaoException {
     public static synchronized int addGeneWithoutEntrezGeneId(CanonicalGene gene) throws DaoException {
         CanonicalGene existingGene = getGene(gene.getHugoGeneSymbolAllCaps());
         gene.setEntrezGeneId(existingGene==null?getNextFakeEntrezId():existingGene.getEntrezGeneId());
-        return addGene(gene);
+        return addOrUpdateGene(gene);
```

Comment:
@sheridancbio yes, but there is no way around this yet. There are many many parts of the code still using the Dao layer. See for example all the methods that use `DaoGeneticAlteration.getGeneticAlterationMap(int, Collection<Long>)` directly or indirectly.